cli/commands/backend/migrate/migrate.go

```diff
@@ -0,0 +1,37 @@
+// Package migrate provides the ability to bootstrap remote state backend.
+package migrate
+
+import (
+	"context"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/internal/errors"
+	"github.com/gruntwork-io/terragrunt/internal/remotestate/backend"
+	"github.com/gruntwork-io/terragrunt/options"
+)
+
+func Run(ctx context.Context, srcPath, dstPath string, opts *options.TerragruntOptions) error {
+	remoteState, err := config.ParseRemoteState(ctx, opts)
+	if err != nil {
+		return err
+	}
+
+	if remoteState == nil {
+		opts.Logger.Debug("Did not find remote `remote_state` block in the config")
+
+		return nil
+	}
+
+	if !opts.ForceBackendMigrate {
+		enabled, err := remoteState.IsVersionControlEnabled(ctx, opts)
```

Comment:
We only need to check this for the source directory, right?

Reply:
Versioning can only be enabled for the entire bucket. We take the bucket settings from the remote state block, where there can only be one.